hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestSnapshotBackgroundServices.java

```diff
@@ -0,0 +1,722 @@
+    OzoneManager newFollowerOM =
+        cluster.getOzoneManager(leaderOM.getOMNodeId());
+    Assertions.assertEquals(leaderOM, newFollowerOM);
+
+    createSnapshotsEachWithNewKeys(newLeaderOM);
+
+    File sstBackupDir = getSstBackupDir(newLeaderOM);
+    File[] files = sstBackupDir.listFiles();
+    Assertions.assertNotNull(files);
+    int numberOfSstFiles = files.length;
+
+    resumeBackupCompactionFilesPruning(newLeaderOM);
+
+    checkIfCompactionBackupFilesWerePruned(sstBackupDir,
+        numberOfSstFiles);
+
+    confirmSnapDiffForTwoSnapshotsDifferingBySingleKey(newLeaderOM);
+  }
+
+  private static void resumeBackupCompactionFilesPruning(
+      OzoneManager ozoneManager) {
+    ozoneManager
+        .getMetadataManager()
+        .getStore()
+        .getRocksDBCheckpointDiffer()
+        .resume();
+  }
+
+  private static void suspendBackupCompactionFilesPruning(
+      OzoneManager ozoneManager) {
+    ozoneManager
+        .getMetadataManager()
+        .getStore()
+        .getRocksDBCheckpointDiffer()
+        .suspend();
+  }
+
+  @Test
+  @DisplayName("testSSTFilteringBackgroundService")
+  public void testSSTFilteringBackgroundService()
+      throws IOException, InterruptedException, TimeoutException {
+    OzoneManager leaderOM = getLeaderOM();
+    OzoneManager followerOM = getInactiveFollowerOM(leaderOM);
+
+    createSnapshotsEachWithNewKeys(leaderOM);
+
+    startInactiveFollower(leaderOM, followerOM, () -> {
+    });
+
+    // Read & Write after snapshot installed.
+    List<String> newKeys = writeKeys(1);
+    readKeys(newKeys);
+
+    OzoneManager newLeaderOM =
+        getNewLeader(leaderOM, followerOM.getOMNodeId(), followerOM);
+    OzoneManager newFollowerOM =
+        cluster.getOzoneManager(leaderOM.getOMNodeId());
+    Assertions.assertEquals(leaderOM, newFollowerOM);
+
+    checkIfSnapshotGetsProcessedBySFS(newLeaderOM);
+
+    confirmSnapDiffForTwoSnapshotsDifferingBySingleKey(
+        newLeaderOM);
+  }
+
+  private void confirmSnapDiffForTwoSnapshotsDifferingBySingleKey(
+      OzoneManager ozoneManager)
+      throws IOException, InterruptedException, TimeoutException {
+    String firstSnapshot = createOzoneSnapshot(ozoneManager,
+        TestSnapshotBackgroundServices.SNAPSHOT_NAME_PREFIX +
+            RandomStringUtils.randomNumeric(10)).getName();
+    String diffKey = writeKeys(1).get(0);
+    String secondSnapshot = createOzoneSnapshot(ozoneManager,
+        TestSnapshotBackgroundServices.SNAPSHOT_NAME_PREFIX +
+            RandomStringUtils.randomNumeric(10)).getName();
+    SnapshotDiffReportOzone diff = getSnapDiffReport(volumeName, bucketName,
+        firstSnapshot, secondSnapshot);
+    Assertions.assertEquals(Collections.singletonList(
+            SnapshotDiffReportOzone.getDiffReportEntry(
+                SnapshotDiffReport.DiffType.CREATE, diffKey, null)),
+        diff.getDiffList());
+  }
+
+  private static void checkIfCompactionBackupFilesWerePruned(File sstBackupDir,
```

Comment:
nit: similar to class definition. 
```suggestion
  private static void checkIfCompactionBackupFilesWerePruned(
      File sstBackupDir,
      int numberOfSstFiles
  ) throws TimeoutException, InterruptedException {
```